### PR TITLE
Update the repository list for sig-policy

### PR DESCRIPTION
**Pull request by gparvin**
I noticed the repo list was old. I removed the archived repos from the list, I could change that to a note if desired.

### CODE REVIEW

**Comment by gparvin** [comment]
/cc @dhaiducek @mprahl 
Let GRC team review first.
/hold

**Review by mprahl — APPROVED** [review]

File: sig-policy/README.md [hunk 1]
```diff
@@ -19,26 +19,30 @@ The following [subprojects][subproject-definition] are owned by sig-policy:
...
 - **Owners:**
-  - [governance-policy-status-sync](https://github.com/open-cluster-management-io/governance-policy-status-sync/blob/main/OWNERS)
-### ocm-kustomize-generator-plugins
-Policy Generator
+  - [policy-collection](https://github.com/open-cluster-management-io/policy-collection/blob/main/OWNERS)
+### policy-generator-plugin
+Policy Generator Plugin
 - **Owners:**
-  - [ocm-kustomize-generator-plugins](https://github.com/open-cluster-management-io/ocm-kustomize-generator-plugins/blob/main/OWNERS)
+  - [policy-collection](https://github.com/open-cluster-management-io/policy-generator-plugin/blob/main/OWNERS)
```
Comment (dhaiducek) [inline sig-policy/README.md]:
```suggestion
  - [policy-generator-plugin](https://github.com/open-cluster-management-io/policy-generator-plugin/blob/main/OWNERS)
```

**Review by dhaiducek — CHANGES_REQUESTED** [review]

File: sig-policy/README.md (hunk 1 above)
Comment (gparvin) [reply sig-policy/README.md]: Thanks, I just fixed this.

**Review by dhaiducek — APPROVED** [review]
LGTM! Thanks for updating this, @gparvin!

**Pull request merged**